rust_src/src/editfns.rs

```diff
@@ -49,3 +49,22 @@ pub fn bobp() -> LispObject {
     let buffer_ref = ThreadState::current_buffer();
     LispObject::from_bool(buffer_ref.pt == buffer_ref.begv)
 }
+
+/// Return t if point is at the beginning of a line.
+#[lisp_fn]
+pub fn bolp() -> LispObject {
+    let buffer_ref = ThreadState::current_buffer();
+    LispObject::from_bool(
+        buffer_ref.pt == buffer_ref.begv || buffer_ref.fetch_byte(buffer_ref.pt_byte - 1) == b'\n',
+    )
+}
+
+/// Return t if point is at the end of a line.
+///`End of a line' includes point being at the end of the buffer.
+#[lisp_fn]
+pub fn eolp() -> LispObject {
+    let buffer_ref = ThreadState::current_buffer();
+    LispObject::from_bool(
+        buffer_ref.pt == buffer_ref.zv() || buffer_ref.fetch_byte(buffer_ref.pt_byte) == b'\n',
```

Comment:
cargo fmt told me about that final comma, but.... does it make sense? I find it a bit strange, no?

Reply:
Agreed, it's a little odd. Perhaps it's a rustfmt bug.

I like that rustfmt is a tool that you just run and it fixes formatting without human input. We could file a bug upstream, but I'd leave the comma for now if rustfmt wants it.

Reply:
I don't think it's a big, it's just consistent in always having a final comma on multi-line lists of expressions. The argument for having the trailing comma in the first place, less churn when an element is added, applies to single arguments as well.